Sources/LiquidAwsS3Driver/LiquidAwsS3StorageConfiguration.swift

```diff
@@ -4,13 +4,47 @@
 //
 //  Created by Tibor Bodecs on 2020. 04. 28..
 //
+import Foundation
 
 struct LiquidAwsS3StorageConfiguration: FileStorageConfiguration {
+    enum S3ConfigurationError: Error {
+        case invalidBucketName
+    }
+
     let key: String
     let secret: String
     let bucket: String
     let region: Region
     let endpoint: String?
+
+    init(key: String, secret: String, bucket: String, region: Region, endpoint: String?) throws {
```

Comment:
I see where this question is directing now.

My thinking is that, I'm not sure if there would ever be a `LiquidAwsS3StorageConfiguration` initialized anywhere other than the `FileStorageConfigurationFactory`, so if it can, it should only initialize if it has a valid bucket name.

So if this validation fails, it will throw and bubble up inside `FileStorageConfigurationFactory.awsS3`'s now throwing function.